torch/testing/_internal/common_modules.py

```diff
@@ -1937,7 +1937,9 @@ def rms_norm_reference_fn(m, p, i):
         normalized_shape = m.normalized_shape
         weight = m.weight
         dims = [ndim - i - 1 for i in range(len(normalized_shape))]
-        result = i * torch.rsqrt(i.pow(2).mean(dim=dims, keepdim=True) + m.eps)
+        upcasted_i = i.float()
```

Comment:
This would fail if `I` is complex and would reduce the precision if `I` is double
```suggestion
        upcasted_i = i.to(dtype=torch.float) if i.dtype==torch.half else i
```

Reply:
i think half doesn't include bf16 right?

Reply:
also, this is just for testcases which should pass